SuperBuild/External_DCMTK.cmake

```diff
@@ -34,11 +34,6 @@ if(NOT DEFINED DCMTK_DIR AND NOT Slicer_USE_SYSTEM_${proj})
   endif()
 
   set(ep_cxx_standard_args)
-  # XXX: On MSVC disable building DCMTK with C++11. DCMTK checks C++11.
```

Comment:
Unless we systematically set `DCMTK_ENABLE_CXX11` on windows, I think this comment is still relevant ?

Reply:
Since `CMAKE_CXX_STANDARD` is available for MSVC, then appears we can make the section below not only in the `if(CMAKE_CXX_STANDARD AND UNIX)` block of code. I can change that.

Reply:
Since following https://github.com/commontk/DCMTK/commit/b47ff36abbb61c8f70d15c08a7c7d8e2299d2083 (`Added support for CMAKE_CXX_STANDARD et al.`)[^1], the CMake variable `CMAKE_CXX_STANDARD` is now fully supported, I suggest we pass the options `CMAKE_CXX_STANDARD`, `CMAKE_CXX_STANDARD_REQUIRED` and `CMAKE_CXX_EXTENSIONS` like we do for the other projects:

https://github.com/Slicer/Slicer/blob/3cf97ce36d6a6ecc72910c316c959a7ff698e5bb/SuperBuild/External_JsonCpp.cmake#L46-L48

This means the workaround originally integrated in https://github.com/Slicer/Slicer/commit/c055f763fb866af39747cfd64eb49a9a04f5d0b6 (ENH: Enable C++11 for external projects and extensions ) is not needed anymore.

[^1]: included in branch [commontk/DCMTK/@patched-DCMTK-3.6.6_20210115/CMake](https://github.com/commontk/DCMTK/tree/patched-DCMTK-3.6.6_20210115/CMake)

Reply:
Pushed update to configure DCMTK like other projects since CMAKE_CXX_STANDARD can be used. This allows to remove the comment for the old workaround.